### ISSUE TITLE
[BUG/ISSUE] Bug fix to LAnthroHgOnly in mercury simulation

### ISSUE THREAD

**Issue by jennyfisher** [geoschem/geos-chem#847]
### Description of the problem
In the mercury code, the logical flag LAnthroHgOnly should set all natural (non-anthropogenic) emissions to zero, but is currently missing the code to zero geogenic emission. Note that some of the natural emissions for mercury are not exclusively dealt with in HEMCO, so while the flag probably isn't best practice it is still somewhat useful for now and we should make it work for now until we have a better solution.

### Description of troubleshooting performed
Harriet Gournia has found that the flag works if HEMCO also used to set NATURAL to False, but we can alternatively fix this in the code.

### GEOS-Chem version
I'm using 12.8.0 - modified to use Viral's version. The problem was reported by Harriet and I'm not sure what version she is using, but I'm assuming a recent standard version.

### Description of modifications
In mercury_mod.f, there is a block of code as follows (starting around line 1488 in my version):
```
          IF ( LAnthroHgOnly ) THEN 
             ! No other emissions
             EHg0_bb(I,J)      = 0e+0_fp
             EHg0_oc(I,J,:)    = 0e+0_fp
             EHg0_snow(I,J,:)  = 0e+0_fp
             IF ( LGTMM ) THEN 
                EHg0_gtm(I,J) = 0e+0_fp
             ELSE 
                EHg0_ln(I,J,:) = 0e+0_fp
                EHg0_vg(I,J)   = 0e+0_fp
                EHg0_so(I,J)   = 0e+0_fp
             ENDIF
          ENDIF
```

This just needs to be modified to the following (new line is the EHg0_geo line, after snow and before GTMM):
```
          IF ( LAnthroHgOnly ) THEN 
             ! No other emissions
             EHg0_bb(I,J)      = 0e+0_fp
             EHg0_oc(I,J,:)    = 0e+0_fp
             EHg0_snow(I,J,:)  = 0e+0_fp
             EHg0_geo(I,J,:)  = 0e+0_fp
             IF ( LGTMM ) THEN 
                EHg0_gtm(I,J) = 0e+0_fp
             ELSE 
                EHg0_ln(I,J,:) = 0e+0_fp
                EHg0_vg(I,J)   = 0e+0_fp
                EHg0_so(I,J)   = 0e+0_fp
             ENDIF
          ENDIF
```

### Log files
n/a

### Software versions
n/a

I haven't tested with this fix, but if we update it I'll see if Harriet can do a quick test.

**Comment by jennyfisher** [comment]
Addition: Harriet is using v13.0.2

**Comment by jennyfisher** [comment]
Embarrassingly I don't really know how to do that - especially as I don't have an updated version of the standard code at the moment (I'm just helping out on this in my capacity as Hg-POPS WG co-chair). I was hoping that since it's just a one-line change (and doesn't impact the standard simulations) it would be easy for someone on GCST to handle it.

Thanks,
Jenny

**Comment by yantosca** [comment]
Thanks Jenny.  I've added the line to the code, as it is a one-liner.  Please see commit 9580101a582418c502526064e83a5f9055a46ac1.  This will ship with version 13.2.1

The reason we prefer pull requests is because it allows us to do a code review before merging, and we can assign it to a Github milestone for documentation.  Also in this case, the code block in question was not at my line 1488 but at line 2762.   As this was a one-liner it is OK, but for more complex updates it gets a little tricky to manually merge code from an older version into a more recent version.   Just for your reference.

**Comment by jennyfisher** [comment]
Thanks, greatly appreciated. I will learn how to do pull requests one day, but I'm not doing much model development work at the moment.

**Comment by yantosca** [comment]
Note, had to issue commit ae7afb8682eeb0f0453d442b44935adfe3b2359d to fix the issue reported in #878.  EHg0_geo is 2-D array but the fix provided in this issue assumed it was a 3-D array.  Sorry for not catching it!  Another reason to use pull requests (it would have caught this).

**Comment by jennyfisher** [comment]
Oops sorry about that!
